res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -1417,17 +1416,15 @@ PioneerDDJSX.tempoResetButton = function(channel, control, value, status, group)
 };
 
 PioneerDDJSX.autoLoopButton = function(channel, control, value, status, group) {
-    if (value) {
-        if (engine.getValue(group, "loop_enabled")) {
-            engine.setValue(group, "reloop_toggle", true);
-        } else {
-            engine.setValue(group, "beatloop_activate", true);
-        }
+    if (engine.getValue(group, "loop_enabled")) {
+        engine.setValue(group, "reloop_toggle", value);
+    } else {
+        engine.setValue(group, "beatloop_activate", value);
```

Comment:
You're right, I didn't see that. But it's strange because it's working as it should. That's why I didn't see it. I will try your solution.